jax/_src/lax/lax.py

```diff
@@ -1413,8 +1413,8 @@ def iota(dtype: DType, size: int) -> Array:
   """
   if config.omnistaging_enabled:
     dtype = dtypes.canonicalize_dtype(dtype)
-    size = core.concrete_or_error(int, size, "size argument of lax.iota")
```

Comment:
I am thinking that here we can use the same code to handle input shapes for primitives. Incidentally, the previous code fails for Poly.